lib/cloud/clients.go

```diff
@@ -547,6 +551,37 @@ func (c *cloudClients) initAWSSession(region string) (*awssession.Session, error
 	return session, nil
 }
 
+func (c *cloudClients) getAWSSessionForRole(ctx context.Context, options SessionOptions) (*awssession.Session, error) {
+	assumeRoler, err := c.GetAWSSTSClient(ctx, SessionOptions{Region: options.Region})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if options.CacheAssumedRole {
```

Comment:
Why we don't always use caching ? Could you add a comment

Reply:
comment added in godoc for the options and I also added a comment inline here to clarify the reason.